sirepo/package_data/template/controls/parameters.py.jinja

```diff
@@ -132,15 +132,15 @@ def run_simulation(correctors):
         for k in values:
             lattice = re.sub('{' + k + '}', str(values[k]), lattice)
         f.write(lattice)
-   
-    success = os.system('madx in.madx > madx.log 2> madx.err') == 0
-    if not success:
+
+    p = subprocess.run(('madx', 'in.madx'), stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+    pkio.write_text('madx.log', p.stdout)
+    pkio.write_text('madx.err', p.stderr)
+    if p.returncode != 0:
         return False
 
-    cmd = 'grep "is lost" madx.log  | wc'
-    res = subprocess.check_output(cmd, shell=True)
-    lost = int(res.decode().strip().split(' ')[0])
-    return not lost == {{ particleCount }}
+    lost = pkio.read_text('madx.log').count('is lost')
```

Comment:
Great. I didn't know about `count`! 

You have the madx.log in p.stdout so you don't need to read anything, just decode to a string with pkcompat.from_bytes.

Reply:
good point 👍

Reply:
```python
lost = pkcompat.locale_str(p.stdout).count('is lost')
return lost != {{ particleCount }}
```

like this ^ ?
didn't see pkcompat.from_bytes in docs, but pkcompat.locale_str seems to do the necessary conversion

Reply:
[from_bytes](https://github.com/radiasoft/pykern/blob/bf5946e20d77d13ceab387c33ac75c59b6579de6/pykern/pkcompat.py#L74) is there but it's hidden well. :(

from_bytes is the correct conversion, because that's what's happening (bytes to str). locale_str converts it to the locale which is not necessarily utf8. Not that this matters exactly, but I think it's better documentation than locale_str which implies there's a localization issue, which there isn't.

I'll create an issue to get rid of py2isms.